evm/src/proof.rs

```diff
@@ -68,6 +68,7 @@ pub struct BlockMetadata {
     pub block_timestamp: U256,
     pub block_number: U256,
     pub block_difficulty: U256,
+    pub block_random: U256,
```

Comment:
Perhaps an overlook, but shouldn't this rather be an `H256` out of consistency with `mix_hash` output?